### PR TITLE
feat: Add support for Timed Clusterization

### PR DESCRIPTION
**Pull request by CarloVarni**
This add possibility of doing clusterization using as well time information

### CODE REVIEW

File: Core/include/Acts/Clusterization/TimedClusterization.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+namespace Acts::Ccl {
+
+template <typename Cell>
+concept HasRetrievableTimeInfo = requires(Cell cell) {
+  { getCellTime(cell) } -> std::same_as<Acts::ActsScalar>;
+};
+
+template <Acts::Ccl::HasRetrievableTimeInfo Cell, std::size_t N>
+struct TimedConnect : public Acts::Ccl::DefaultConnect<Cell, N> {
+  Acts::ActsScalar timeTollerance{std::numeric_limits<Acts::ActsScalar>::max()};
```
Comment (gagnonlg) [inline Core/include/Acts/Clusterization/TimedClusterization.hpp:25]: -> timeTolerance

File: Core/include/Acts/Clusterization/TimedClusterization.hpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  TimedConnect() = default;
+  TimedConnect(Acts::ActsScalar time);
+  TimedConnect(Acts::ActsScalar time, bool conn)
+    requires(N == 2);
```
Comment (gagnonlg) [inline Core/include/Acts/Clusterization/TimedClusterization.hpp:30]: Why template on N then require N == 2?

File: Core/include/Acts/Clusterization/TimedClusterization.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  { getCellTime(cell) } -> std::same_as<Acts::ActsScalar>;
+};
+
+template <Acts::Ccl::HasRetrievableTimeInfo Cell, std::size_t N>
+struct TimedConnect : public Acts::Ccl::DefaultConnect<Cell, N> {
+  Acts::ActsScalar timeTollerance{std::numeric_limits<Acts::ActsScalar>::max()};
+
+  TimedConnect() = default;
+  TimedConnect(Acts::ActsScalar time);
+  TimedConnect(Acts::ActsScalar time, bool conn)
```
Comment (gagnonlg) [inline Core/include/Acts/Clusterization/TimedClusterization.hpp:29]: conn -> commonCorner (like the default case)

File: Core/include/Acts/Clusterization/TimedClusterization.ipp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+namespace Acts::Ccl {
+
+template <Acts::Ccl::HasRetrievableTimeInfo Cell, std::size_t N>
+TimedConnect<Cell, N>::TimedConnect(Acts::ActsScalar time)
+    : timeTollerance(time) {}
+
+template <Acts::Ccl::HasRetrievableTimeInfo Cell, std::size_t N>
+TimedConnect<Cell, N>::TimedConnect(Acts::ActsScalar time, bool conn)
```
Comment (gagnonlg) [inline Core/include/Acts/Clusterization/TimedClusterization.ipp:16]: conn -> commonCorner (like the default case)

**Review by gagnonlg — CHANGES_REQUESTED** [review]
Nice work @CarloVarni , I only have a few comments

File: Core/include/Acts/Clusterization/TimedClusterization.hpp (hunk 2 above)
Comment (CarloVarni) [reply Core/include/Acts/Clusterization/TimedClusterization.hpp:30]: This also supports the 1D case. the first two constructors are compatible with both 1D and 2D case. The constructor with the `bool` only works for 2D case

File: Core/include/Acts/Clusterization/TimedClusterization.hpp (hunk 1 above)
Comment (CarloVarni) [reply Core/include/Acts/Clusterization/TimedClusterization.hpp:25]: ops... this is italian sneaking in

File: Core/include/Acts/Clusterization/TimedClusterization.hpp (hunk 3 above)
Comment (CarloVarni) [reply Core/include/Acts/Clusterization/TimedClusterization.hpp:29]: kk

File: Core/include/Acts/Clusterization/TimedClusterization.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Clusterization/TimedClusterization.hpp]: 🇮🇹

**Review by gagnonlg — APPROVED** [review]
Nicely done

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [6cea6b2ddf6fb6c47d2fc9495463643fbbee7890]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44329831) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8260958)!

**Pull request merged**